gapic-generator/lib/gapic/path_pattern/pattern.rb

```diff
@@ -88,6 +88,20 @@ def to_regex_str
         regex_str
       end
 
+      ##
+      # Converts the PathPattern into a regex string that matches a whole field
+      # - adds markers for the beginning and end of the string
+      # - adds handling of an optional tail `/` if needed
+      # @return [String]
+      def to_field_regex_str
+        tail_slash_accept = if ends_with_double_star_pattern?
+                              ""
+                            else
+                              "/?"
+                            end
```

Comment:
Ternary operator is probably just fine here:

```ruby
tail_slash_accept = ends_with_double_star_pattern? ? "" : "/?"
```

Reply:
:+1: